src/main/java/com/spikes2212/command/genericsubsystem/smartmotorcontrollersubsystem/CTRESmartMotorControllerGenericSubsystem.java

```diff
@@ -0,0 +1,124 @@
+package com.spikes2212.command.genericsubsystem.smartmotorcontrollersubsystem;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.IFollower;
+import com.ctre.phoenix.motorcontrol.can.BaseMotorController;
+import com.spikes2212.command.DashboardedSubsystem;
+import com.spikes2212.control.FeedForwardSettings;
+import com.spikes2212.control.PIDSettings;
+import com.spikes2212.control.TrapezoidProfileSettings;
+import edu.wpi.first.wpilibj.motorcontrol.MotorController;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+
+import java.util.List;
+
+/**
+ * A subsystem which consists of a CTRE motor controller that can run PID loops and additional
+ * CTRE motors that follow it.
+ *
+ * @param <T> the type of the motor controller on which the loop is run
+ *
+ * @author Yoel Perman Brilliant
+ * @see DashboardedSubsystem
+ * @see SmartMotorControllerSubsystem
+ * @see BaseMotorController
+ * @see MotorController
+ */
+public class CTRESmartMotorControllerGenericSubsystem<T extends BaseMotorController & MotorController>
+        extends DashboardedSubsystem implements SmartMotorControllerSubsystem {
+
+    /**
+     * The slot on the motor controller on which the loop is run.
+     */
+    private static final int SLOT = 0;
+
+    /**
+     * The motor controller which runs the loops.
+     */
+    protected final T master;
+
+    /**
+     * Additional motor controllers that follow the master.
+     */
+    protected final List<? extends IFollower> slaves;
+
+    /**
+     * Constructs a new instance of {@link CTRESmartMotorControllerGenericSubsystem}.
+     *
+     * @param namespaceName the name of the subsystem's namespace
+     * @param master the motor controller which runs the loops
+     * @param slaves additional motor controllers that follow the master
+     */
+    public CTRESmartMotorControllerGenericSubsystem(String namespaceName, T master,
+                                                    IFollower... slaves) {
+        super(namespaceName);
+        this.master = master;
+        this.slaves = List.of(slaves);
+        this.slaves.forEach(s -> s.follow(master));
+    }
+
+    /**
+     * Adds any data or commands to the {@link Shuffleboard}.
```

Comment:
Technically it puts it on the network tables?

Reply:
I changed it, but the documentation in MotoredGenericSubsystem says "Adds any commands or data from this subsystem to the dashboard" so we should probably decide on a consistent rule